sklearn/utils/tests/test_validation.py

```diff
@@ -1153,3 +1153,27 @@ def test_check_fit_params(indices):
         result['sparse-col'],
         _safe_indexing(fit_params['sparse-col'], indices_)
     )
+
+
+def test_check_sparse_pandas_default():
+    pd = pytest.importorskip("pandas")
+    sp_mat = _sparse_random_matrix(10, 3)
+
+    sdf = pd.DataFrame.sparse.from_spmatrix(sp_mat)
+    result = check_array(sdf, accept_sparse=True)
+
+    # by default pandas converts to coo
+    assert sp.issparse(result)
+    assert result.format == 'coo'
+    assert_allclose_dense_sparse(sp_mat, result)
+
+
+@pytest.mark.parametrize('sp_format', ['csr', 'csc', 'coo', 'bsr'])
```

Comment:
Maybe you can add the `True` option here to merge both tests?

Also as usual, a short comment describing the test would help for future us